exercise-integration-testing-5-53/test/spread-the-code-for-task-10/when-event-is-valid.test.js

```diff
@@ -0,0 +1,270 @@
+      //Arrange
+  
+      //Act
+  
+      const eventToAdd = {
+        category: `${helper(5)}`,
+        temperature: 20,
+        name: "Thermostat-temperature-new", //This must be unique
+        color: "Green",
+        weight: "80 gram",
+        status: "active",
+      };
+  
+      console.log(eventToAdd);
+      const config = {
+        validateStatus: () => true,
+      };
+      const urlFetchSensorEvent = `http://localhost:${port}/sensor-events`;
+      const urlGetEventByCategory = `http://localhost:${port}/sensor-events/${eventToAdd.category}/${eventToAdd.name}`;
+  
+      await axios.post(urlFetchSensorEvent, eventToAdd, config);
+      const eventFromDb = await axios.get(urlGetEventByCategory);
+      console.log("retrive data ", eventFromDb.data.length);
+      //Assert
+      expect(eventFromDb.status).toBe(200);
+      expect(eventFromDb.data.length).toBe(1); // One obj from the last test and one is from the current test.
+    });
+  
+    // ✅ TASK: Test that querying the GET:/sensor-events route, it returns the right events when multiple events exist
+    // 💡 TIP: Ensure that all the relevant events were returned
+  
+    test("When new valid event is send, should retrievable the data from the DB when muliple events exist", async () => {
+      //Arrange
+  
+      const category = helper(5);
+  
+      const firstEventToAdd = {
+        category: `${category}`,
+        temperature: 20,
+        name: "Thermostat-temperature", //This must be unique
+        color: "Green",
+        weight: "80 gram",
+        status: "active",
+      };
+  
+      const secondEventToAdd = {
+        category: `${category}`,
+        temperature: 25,
+        name: "Thermostat-temperature-new-one", //This must be unique
+        color: "Green",
+        weight: "80 gram",
+        status: "active",
+      };
+  
+      //Act
+  
+      const config = {
+        validateStatus: () => true,
+      };
+      const urlFetchSensorEvent = `http://localhost:${port}/sensor-events`;
+      const urlGetEventByCategory = `http://localhost:${port}/sensor-events/${firstEventToAdd.category}/${firstEventToAdd.name}`;
+  
+      await axios.post(urlFetchSensorEvent, firstEventToAdd, config);
+      await axios.post(urlFetchSensorEvent, secondEventToAdd, config);
+  
+      const eventFromDb = await axios.get(urlGetEventByCategory);
+      //Assert
+      expect(eventFromDb.status).toBe(200);
+      expect(eventFromDb.data.length).toBe(2);
+    });
+  
+    // ✅ TASK: Test that querying for /sensor-events route and sorting by the field 'name', the results are indeed sorted
+    // 💡 TIP: Each test should be independent and might run alone without others, don't count on data (events) from other tests
+  
+    test("When new 3 valid events is send, should return the events while them sort by name", async () => {
+      //Arrange
+      const category = helper(5);
+  
+      const firstEventToAdd = {
```

Comment:
Use helper to factor 3 objects, it's a bit too long. Just make it clear what the helper is doing and pass params for important data that affects the test result: testHelper.factorEvent({category:something})